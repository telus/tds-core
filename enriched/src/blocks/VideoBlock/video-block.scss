.video-container-block {
	position: relative;
	padding-bottom: 53.10%;
	padding-top: 25px;
	height: 0;
	margin-bottom: 33px;
	iframe {
		position: absolute;
		top: 0;
		left: 0;
		width: 100%;
		height: 100%;
	}
}
