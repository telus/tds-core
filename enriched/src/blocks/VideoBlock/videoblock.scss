.video-container {
	position: relative;
	padding-bottom: 53.10%;
	padding-top: 25px;
	height: 0;
	margin-bottom: 33px;
}
.video-container iframe {
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
}
