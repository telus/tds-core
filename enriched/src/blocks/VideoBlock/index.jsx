import React from 'react';
import PropTypes from 'prop-types';

if (process.env.BROWSER) {
  require('./video-block.scss');
}

const VideoBlock = (props) => {
  const { caption, subtext, videoUrl } = props;

  return (
    <section className="video-block">
      <div className="container container--limited-width">
        <div className="grid-row">
          <div className="medium-8">
            <div className="video-container-block">
              <iframe allowFullScreen="" frameBorder="0" src={videoUrl} title={caption} />
            </div>
            { caption && <p className="text text--medium">
              <strong>{caption}: </strong>
              {subtext}
            </p>}
          </div>
        </div>
      </div>
    </section>
  );
};

VideoBlock.propTypes = {
  caption: PropTypes.string,
  subtext: PropTypes.string,
  videoUrl: PropTypes.string
};

VideoBlock.defaultProps = {
  subtext: '',
  caption: '',
  videoUrl: ''
};

export default VideoBlock;
