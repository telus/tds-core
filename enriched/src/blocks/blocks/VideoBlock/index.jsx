import React from 'react';

if ( process.env.BROWSER ) {
  require('./videoblock.scss');
}

const VideoBlock = (props) => {
  const { title, subtext, videoUrl } = props;

  return (
    <div className="videoblock">
      <div className="container container--limited-width">
        <div className="grid-row">
          <div className="medium-8">
            <div className="video-container">
              <iframe allowFullScreen="" frameBorder="0" src={videoUrl} title={title}></iframe>
            </div>
            { title && <p className="text text--medium">
              <strong>{title}: </strong>
              {subtext}
            </p>}
          </div>
        </div>
      </div>
    </div>
  );
};

VideoBlock.propTypes = {
  title: React.PropTypes.string,
  subtext: React.PropTypes.string,
  videoUrl: React.PropTypes.string
};

VideoBlock.defaultProps = {
  subtext: '',
  title: '',
  videoUrl: ''
};

export default VideoBlock;
