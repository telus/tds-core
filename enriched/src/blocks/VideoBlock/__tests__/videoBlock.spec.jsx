import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import VideoBlock from '../';

describe('VideoBlock', () => {
  const data = { caption: 'Foo', subtext: 'Bar', videoUrl: 'https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1' };

  it('matches snapshot', () => {
    const wrapper = mount(<VideoBlock {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render caption', () => {
    const wrapper = mount(<VideoBlock {...data} />);
    expect(wrapper.find('strong')).toBeDefined();
    console.log(wrapper.find('strong').first().text())
    expect(wrapper.find('strong').text()).toBe('Foo: ');
  });

  it('should render subtext', () => {
    const wrapper = mount(<VideoBlock {...data} />);
    expect(wrapper.find('p').first().text()).toContain('Bar');
  });

  it('should render without title or subtext', () => {
    const wrapper = mount(<VideoBlock videoUrl={data.videoUrl} />);
    expect(wrapper.find('.text--medium').exists()).toEqual(false);
  });

  it('should render an iframe with video', () => {
    const wrapper = mount(<VideoBlock videoUrl={data.videoUrl} />);
    expect(wrapper.find('iframe').first().prop('src')).toEqual('https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1');
  });
});
