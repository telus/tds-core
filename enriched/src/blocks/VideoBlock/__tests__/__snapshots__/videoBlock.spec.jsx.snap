// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoBlock matches snapshot 1`] = `
<VideoBlock
  caption="Foo"
  subtext="Bar"
  videoUrl="https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1"
>
  <section
    className="video-block"
  >
    <div
      className="container container--limited-width"
    >
      <div
        className="grid-row"
      >
        <div
          className="medium-8"
        >
          <div
            className="video-container-block"
          >
            <iframe
              allowFullScreen=""
              frameBorder="0"
              src="https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1"
              title="Foo"
            />
          </div>
          <p
            className="text text--medium"
          >
            <strong>
              Foo
              : 
            </strong>
            Bar
          </p>
        </div>
      </div>
    </div>
  </section>
</VideoBlock>
`;
